### PR TITLE
feat: introduce next generation ADC

### PR DESCRIPTION
**Pull request by bzp2010**
### Description

Introducing Next Generation ADC.

Currently, ADC is implemented in a rudimentary language, Golang, with a lack of expressive syntax that hampers the implementation and maintenance of its functionality, and this state of affairs has to be changed. CLI tools are simply not a performance-sensitive scenario, and the use of Golang is not justified.

Scripting languages strike a balance between performance and language features. [NodeJS](https://nodejs.org/en) uses the [V8 JS engine](https://v8.dev/) to perform JIT compilation with respectable performance, and TypeScript provides the best language features that can be compiled to JavaScript, so the next generation of ADCs uses TypeScript + NodeJS stack.

Thanks to the [SEA](https://nodejs.org/api/single-executable-applications.html) of NodeJS, ADC can still be compiled into a single executable file and run on major operating systems (Linux, Windows, macOS) and major CPU instruction (AMD64, ARM64), and it provides pre-built binaries for all six platforms. In the future, it can also be published to npm repositories and accessed using npx, which will allow users to use ADC on more platforms that don't support SEA compilation at the moment, and will be easy to integrate with the CI/CD pipeline.

We use Nx as the build toolchain, and manage packages in a modular way, so that ADC's backend and converters will be able to evolve independently and be built with major releases. This provides greater modularity than the "primitive" package mechanism in Golang.

On the private dev branch, it's pretty much done by me personally, there have been hundreds of commits, commits from other developers have been reviewed or completely rewritten in the current commit. Too many commits logged could prevent this PR from being readable, so they are collapsed into a single PR.

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] ~~I have verified that this change is backward compatible~~ Answer is NO

### CODE REVIEW

**Review by pottekkat — APPROVED** [review]
🎉

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      # Run E2E tests
+      - name: Run E2E tests
+        run: npx nx run backend-apisix:e2e
+
+  api7:
+    runs-on: ubuntu-latest
+    if: contains(github.event.pull_request.labels.*.name, 'test/api7')
+    strategy:
+      matrix:
+        version: [3.2.13.0, 3.2.14.0]
```
Comment (juzhiyuan) [inline .github/workflows/e2e.yaml:41]: Should we use 3.2.13.8 or .0?

**Review by juzhiyuan — APPROVED** [review]
LGTM

File: .github/workflows/e2e.yaml (hunk 1 above)
Comment (bzp2010) [reply .github/workflows/e2e.yaml:41]: sounds good

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**